routing.go

```diff
@@ -109,15 +109,15 @@ func (ps *proxyRouting) fetch(ctx context.Context, key string) (rb []byte, err e
 	}
 	defer resp.Body.Close()
 
-	if resp.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf("routing/get RPC returned unexpected status: %s", resp.Status)
-	}
-
 	rb, err = io.ReadAll(resp.Body)
```

Comment:
we now read the entire body on HTTP error
would it be possible to wrap this in LimitedReader so `rb`  and error if it is bigger than the max size limit of IPNS Record ([10KiB](https://specs.ipfs.tech/ipns/ipns-record/#record-size-limit)), just to be safe?